src/fit.jl

```diff
@@ -337,4 +339,94 @@ function calculate_multilocal_masks_and_ranges(windows, obs_wls, synthesis_wls,
     obs_wl_mask, synth_wl_mask, multi_synth_wls
 end
 
+"""
+    linelist_neighbourhood_indices(linelist, line_buffer)
+
+Group lines together such that no two lines are closer than twice the value of `line_buffer`.
+
+# Arguments:
+- `linelist`: A vector of [`Line`](@ref)s (see [`read_linelist`](@ref), 
+   [`get_APOGEE_DR17_linelist`](@ref), and [`get_VALD_solar_linelist`](@ref)).
+- `line_buffer`: half the minimum separation (in Å) of lines in a group
+
+# Returns
+A vector of vectors, where each inner vector contains the indices of lines in a group.
+"""
+function linelist_neighbourhood_indices(linelist, line_buffer)
```

Comment:
That's correct. In practice there are usually only a few close lines together, even for Solar twins work, because astronomers pre-selected which lines they prefer. For the Solar twin line list I was testing there were ~100 lines which were split into 3 lists. But this is really a hack to make things work -- I'd love a different way to handle this!

Reply:
I think it’s ok for now if we output a warning when there’s a ton of groups

Reply:
Do you have a feeling for what measure constitutes a 'ton' here? >10 groups?